services/src/main/java/org/keycloak/services/clientregistration/oidc/DescriptionConverter.java

```diff
@@ -270,6 +274,10 @@ public static OIDCClientRepresentation toExternalResponse(KeycloakSession sessio
         response.setBackchannelLogoutSessionRequired(config.isBackchannelLogoutSessionRequired());
         response.setBackchannelLogoutSessionRequired(config.getBackchannelLogoutRevokeOfflineTokens());
 
+        if (config.getBackchannelTokenDeliveryMode() != null) {
+            response.setBackchannelTokenDeliveryMode(config.getBackchannelTokenDeliveryMode());
+        }
+
         List<ProtocolMapperRepresentation> foundPairwiseMappers = PairwiseSubMapperUtils.getPairwiseSubMappers(client);
         SubjectType subjectType = foundPairwiseMappers.isEmpty() ? SubjectType.PUBLIC : SubjectType.PAIRWISE;
         response.setSubjectType(subjectType.toString().toLowerCase());
```

Comment:
Makes sense for me.

Reply:
@mposolda @pedroigor I agree with you. How about the followings?

- add CIBA Grant switch the same as Device Grant.
- in `DescriptionConverter.getOIDCGrantTypes`, convert CIBA Grant switch setting to the value of `grant_type`
- in `DescriptionConverter.toInternal`, convert the value of `grant_type` to CIBA Grant switch setting.

The current `DescriptionConverter.toInternal`, Resource Owner Password Credentials Grant and Client Credentials Grant are converted their value of `grant_type` to their switch settings (Direct Access, Service Account) while others are determined based on `response_type`. IMO we might reconsider the logic so that it could be treated as the other JIRA ticket.

Reply:
+1. The OIDC client registration specification ( https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata ) mentions this: `The following table lists the correspondence between response_type values that the Client will use and grant_type values that MUST be included in the registered grant_types list .`

So in theory, we may throw the exception during client registration in case that `grant_types` and `response_types` are incompatible? For example if `response_types` is just `authorization_code`, but the `grant_types` has also grants for implicit/hybrid flow like `code id_token` etc.

But that is maybe something for the future? For CIBA, there is no corresponding response type AFAIK (Similarly like for example for Resource Owner Password Credentials or Client Credentials Grant). Hence in this PR, I would just convert grant type for CIBA to the corresponding switch of client as you suggested.

Reply:
@mposolda 
> So in theory, we may throw the exception during client registration in case that grant_types and response_types are incompatible? For example if response_types is just authorization_code, but the grant_types has also grants for implicit/hybrid flow like code id_token etc.

Yes, in theory, we thrown an exception in this case.

For CIBA, it is not relevant to `response_type` as you mentioned. Therefore, I would like to convert `grant_type` and the switch setting.